Tumusime14/Certificate-Manager---David
# Feature/Task-3-KAN-46 Button for input and doc priview

### PR Description
This pull request introduces a new feature that adds data through form and displays the document before sending it.
Ensuring security and preventing incorrect submission of data into the app.

### Changes made:

1. A "New Certificate" button was added to the interface.
2. I implemented a form allowing users to input and submit certificate details.
3. The form includes fields for Supplier, Certificate Type, Valid From, Valid To, and a PDF upload with a preview.
4. Form validation is considered for required fields and file type, ensuring only PDFs are uploaded and non-compliant entries are handled appropriately.
5. **Implemented state management to store and display submitted data in the existing table.**

### Checklist

- [x] Added a "New Certificate" button to the interface.
- [x] Implemented a form that allows users to input and submit new certificate details.
- [x] The form includes fields for Supplier, Certificate Type, Valid From, Valid To, and a PDF upload with a preview.
- [x] Included validation for required fields and file type.

[Jira Ticket ](https://task-force-pro-dccs.atlassian.net/jira/software/projects/KAN/boards/1?assignee=712020%3Ab9bb79c7-4352-4d17-9af2-5c36278b1670%2C712020%3A5d939816-4bad-4fed-be90-a36cfb42d87f&selectedIssue=KAN-46)

### How to Test

- Navigate to the dropdown items and press "Example 1" on top of the table.
- Press the "New Certificate" button.
- Test all form fields: Supplier, Certificate Type, Valid From, Valid To.
- Upload a PDF file and verify that the preview is displayed.
- Attempt to upload a non-PDF file and ensure proper error handling.

### Updated screenshot


![Form input fields and doc preview](https://github.com/user-attachments/assets/7856a4d1-b376-4caa-9ab6-6a4d8f110309)


![Table saved data](https://github.com/user-attachments/assets/847fec70-ac34-409a-a4b7-8441cd87ef7f)
